### PR TITLE
Sttp3 migration

### PR DESCRIPTION
**Pull request by bwiercinski**

### CODE REVIEW

File: oauth2/src/main/scala/com/ocadotechnology/sttp/oauth2/AuthorizationCode.scala [hunk 1]
```diff
@@ -3,23 +3,26 @@ package com.ocadotechnology.sttp.oauth2
...
+      .addParam("response_type", "code")
+      .addParam("client_id", clientId)
+      .addParam("redirect_uri", redirectUri)
+      .addParam("state", state)
+      .addParam("scope", scopes.mkString(" "))
 
   private def prepareLogoutLink(baseUri: Uri, clientId: String, redirectUri: String): Uri =
-    baseUri.path("logout").param("client_id", clientId).param("redirect_uri", redirectUri)
+    baseUri
+      .addPath("logout")
```
Comment (bwiercinski) [inline oauth2/src/main/scala/com/ocadotechnology/sttp/oauth2/AuthorizationCode.scala:23]: here `path` was used (which is now `withPath`), but I guess we should use `addPath` just like in the `prepareLoginLink` function

File: oauth2/src/main/scala/com/ocadotechnology/sttp/oauth2/UserInfoProvider.scala [hunk 2]
```diff
@@ -20,30 +19,31 @@ object UserInfoProvider {
...
-      .post(baseUrl.path(List("openid", "userinfo")))
-      .header("Authorization", s"Bearer $accessToken")
-      .response(asString)
-      .send()
+    implicit sttpBackend: SttpBackend[F, Any]
+  ): F[UserInfo] =
+    sttpBackend
+      .send {
+        basicRequest
+          .post(baseUrl.withPath("openid", "userinfo"))
```
Comment (bwiercinski) [inline oauth2/src/main/scala/com/ocadotechnology/sttp/oauth2/UserInfoProvider.scala:27]: should we consider this?
```suggestion
          .post(baseUrl.addPath("openid", "userinfo"))
```

File: oauth2/src/main/scala/com/ocadotechnology/sttp/oauth2/AuthorizationCode.scala (hunk 1 above)
Comment (kubukoz) [reply oauth2/src/main/scala/com/ocadotechnology/sttp/oauth2/AuthorizationCode.scala:23]: yeah I believe so

File: oauth2/src/main/scala/com/ocadotechnology/sttp/oauth2/UserInfoProvider.scala (hunk 2 above)
Comment (kubukoz) [reply oauth2/src/main/scala/com/ocadotechnology/sttp/oauth2/UserInfoProvider.scala:27]: It'll work, I think it's fine.

**Review by kubukoz — APPROVED** [review]

File: oauth2/src/main/scala/com/ocadotechnology/sttp/oauth2/AuthorizationCode.scala [hunk 3]
```diff
@@ -54,13 +59,15 @@ object AuthorizationCode {
     clientSecret: Secret[String],
     scopeOverride: ScopeSelection
   )(
-    implicit backend: SttpBackend[F, Nothing, NothingT]
+    implicit sttpBackend: SttpBackend[F, Any]
```
Comment (tplaskowski) [inline oauth2/src/main/scala/com/ocadotechnology/sttp/oauth2/AuthorizationCode.scala:62]: I liked the `backed` name, we know that it is `sttpBackend` from the type.

File: oauth2/src/main/scala/com/ocadotechnology/sttp/oauth2/AuthorizationCode.scala (hunk 1 above)
Comment (majk-p) [reply oauth2/src/main/scala/com/ocadotechnology/sttp/oauth2/AuthorizationCode.scala:23]: I wouldn't be so sure. It seems we (unfortunately) currently rely on this behavior. Internally our baseUri is `https://internal-sso-service-host/oauth2` and call to `path` would make it `https://internal-sso-service-host/logout`. Switching to `addPath` changes this to `https://internal-sso-service-host/oauth2/logout` - this endpoint doesn't exist.
This of course doesn't look too good, but let's leave it as is for now not to brake current use cases.

This issue should be addressed by https://github.com/ocadotechnology/sttp-oauth2/issues/9 

@bwiercinski can you please restore that?

File: oauth2/src/main/scala/com/ocadotechnology/sttp/oauth2/AuthorizationCode.scala (hunk 3 above)
Comment (bwiercinski) [reply oauth2/src/main/scala/com/ocadotechnology/sttp/oauth2/AuthorizationCode.scala:62]: i can consider changing the variable name, but everywhere. presently `backend` and `sttpBackend` are 50/50

Comment (bwiercinski) [reply oauth2/src/main/scala/com/ocadotechnology/sttp/oauth2/AuthorizationCode.scala:62]: inside sttp code `backend` name is used, so i will stick to that

Comment (tplaskowski) [reply oauth2/src/main/scala/com/ocadotechnology/sttp/oauth2/AuthorizationCode.scala]: Yup, that was the intention. Thanks 😉

File: oauth2/src/main/scala/com/ocadotechnology/sttp/oauth2/AuthorizationCode.scala (hunk 1 above)
Comment (bwiercinski) [reply oauth2/src/main/scala/com/ocadotechnology/sttp/oauth2/AuthorizationCode.scala]: done, i've kept the old behavior

File: oauth2/src/main/scala/com/ocadotechnology/sttp/oauth2/UserInfoProvider.scala (hunk 2 above)
Comment (bwiercinski) [reply oauth2/src/main/scala/com/ocadotechnology/sttp/oauth2/UserInfoProvider.scala:27]: using `withPath` to keep old behavior

**Review by tplaskowski — APPROVED** [review]

File: oauth2/src/main/scala/com/ocadotechnology/sttp/oauth2/AuthorizationCode.scala (hunk 1 above)
Comment (kubukoz) [reply oauth2/src/main/scala/com/ocadotechnology/sttp/oauth2/AuthorizationCode.scala]: big oof

**Pull request merged**